### PR TITLE
Refactor Telemetry SDK resource detector description

### PR DESCRIPTION
**Pull request by pellared**
I find the current description not well structured and also it might be a little misleading.

Related to https://github.com/open-telemetry/opentelemetry-specification/issues/3382

### CODE REVIEW

File: semantic_conventions/resource/telemetry.yaml [hunk 1]
```diff
@@ -10,6 +10,13 @@ groups:
         requirement_level: required
         brief: >
           The name of the telemetry SDK as defined above.
+        note: |
+          The OpenTelemetry SDK MUST set the `telemetry.sdk.name` attribute to `opentelemetry`.
+          If another SDK, like a fork or a vendor-provided implementation, is used, this SDK MUST set the
+          `telemetry.sdk.name` attribute to the fully-qualified class or module name of this SDK's main entry point
```
Comment (carlosalberto) [inline semantic_conventions/resource/telemetry.yaml:16]: Non-blocker comment: Do we need to request a fully qualified name here? I imagine an alternative implementation, which would need to set this to `io.another.system.Implementation` as opposed to `anotherimplementation` (yes, I saw the "another suitable identifier" part too, but still ;) ).

**Review by carlosalberto — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Review by reyang — APPROVED** [review]

File: semantic_conventions/resource/telemetry.yaml [hunk 2]
```diff
@@ -10,6 +10,13 @@ groups:
         requirement_level: required
         brief: >
           The name of the telemetry SDK as defined above.
+        note: |
+          The OpenTelemetry SDK MUST set the `telemetry.sdk.name` attribute to `opentelemetry`.
+          If another SDK, like a fork or a vendor-provided implementation, is used, this SDK MUST set the
```
Comment (joaopgrassi) [inline semantic_conventions/resource/telemetry.yaml:15]:
```suggestion
          If another SDK, such as a fork or a vendor-provided implementation is used, this SDK MUST set the
```

File: semantic_conventions/resource/telemetry.yaml [hunk 3]
```diff
@@ -10,6 +10,13 @@ groups:
         requirement_level: required
         brief: >
           The name of the telemetry SDK as defined above.
+        note: |
+          The OpenTelemetry SDK MUST set the `telemetry.sdk.name` attribute to `opentelemetry`.
```
Comment (joaopgrassi) [inline semantic_conventions/resource/telemetry.yaml:14]: Another alternative would be:

```suggestion
          SDKs provided by the OpenTelemetry project MUST set the `telemetry.sdk.name` attribute to `opentelemetry`.
```

To emphasize that "oficial" OTel SDKs must do that.

File: semantic_conventions/resource/telemetry.yaml (hunk 2 above)
Comment (pellared) [reply semantic_conventions/resource/telemetry.yaml:15]: We can do it in a separate PR (similar comment https://github.com/open-telemetry/opentelemetry-specification/pull/3392#discussion_r1166961838)

File: semantic_conventions/resource/telemetry.yaml (hunk 3 above)
Comment (pellared) [reply semantic_conventions/resource/telemetry.yaml:14]: We can do it in a separate PR (similar comment https://github.com/open-telemetry/opentelemetry-specification/pull/3392#discussion_r1166961838)

**Comment by carlosalberto** [comment]
@joaopgrassi Will merge this PR the way it is now. If you feel the wording regarding the default implementation should be 'tuned', let me know so I open a PR so we discuss it (or else open the PR yourself).

**Pull request merged**